safe_core/src/client/mock/routing.rs

```diff
@@ -108,10 +122,14 @@ impl Routing {
             proxy_node_name: new_rand::random(),
         };
 
+        let bls_sk = BlsSecretKey::random();
```

Comment:
I think we should have the BLS secret key in `mock_routing::FullId`? 
I.e. we can probably replace this with
```suggestion
        let bls_sk = id.map(|id| id.bls_key()).unwrap_or_else(BlsSecretKey::random);
```